File: simple/javavulnweb/src/main/java/com/vulnweb/api/VulnInterface.java
package com.vulnweb.api;

/**
 * 独立接口定义文件
 * 定义漏洞执行接口，用于接口实现类调用
 */
public interface VulnInterface {
    /**
     * 执行命令
     * @param cmd 命令参数
     * @return 执行结果
     */
    String executeCmd(String cmd);

    /**
     * 执行代码
     * @param script 脚本参数
     * @return 执行结果
     */
    String executeCode(String script);

    /**
     * 执行SSRF
     * @param url URL参数
     * @return 执行结果
     */
    String executeSsrf(String url);
}

File: simple/pythonvulnweb/interface/vuln_interface.py
from abc import ABC, abstractmethod

class VulnInterface(ABC):
    @abstractmethod
    def execute(self, param):
        # 抽象方法：子类必须实现，用于执行漏洞操作
        # 参数param可以是命令、代码或URL，根据漏洞类型不同
        pass
    
    @abstractmethod
    def hardcoded_execute(self):
        # 抽象方法：子类必须实现，用于执行硬编码漏洞操作
        pass